src/bot/api.rs

```diff
@@ -41,15 +42,14 @@ impl Bot {
         )
     }
 
-    pub fn forward_message<C, F, M>(
-        &self,
+    pub fn forward_message<'a, C, F, M>(
+        &'a self,
         chat_id: C,
-        from_chat_id: F,
+        from_chat_id: C,
         message_id: M,
     ) -> ForwardMessage
     where
-        C: Into<ChatId>,
-        F: Into<ChatId>,
+        C: Into<Cow<'a, ChatId>>,
         M: Into<i32>,
```

Comment:
1. `chat_id` and `from_chat_id` should use different generics
2. `Cow` for `ChatId` is useless. It will be either `Borrowed(&'a ChatId)` or `Owned(ChatId)`, thats almost the same! (in both variants there can be `ChatId::ChannelUsername` that contains String)

To make `Cow` useful on `ChatId` we should create Owned/Borrowed variants of it:
```rust
pub enum ChatId<'a> {
    Id(i32),
    ChannelUsername(&'a str), // You can't actually create custom DST
}

pub enum OwnedChatId {
    Id(i64),
    ChannelUsername(String),
}

impl ToOwned for ChatId<'_> {
    type Owned = OwnedChatId;

    // ...
}
```
OR
use `Cow` in ChatId:

```rust
pub enum ChatId<'a> {
    Id(i64),
    ChannelUsername(Cow<'a, str>),
}

impl From<&str> for ChatId<'_> { /* ... */}
impl From<String> for ChatId<'_> { /* ... */}
```